webapp/src/ts/services/performance.service.ts

```diff
@@ -0,0 +1,92 @@
+import { Inject, Injectable } from '@angular/core';
+import { DOCUMENT } from '@angular/common';
+
+import { TelemetryService } from '@mm-services/telemetry.service';
+import { AuthService } from '@mm-services/auth.service';
+
+@Injectable({
+  providedIn: 'root'
+})
+export class PerformanceService {
+  private trackPerformance = false;
+  private readonly CAN_TRACK_PERFORMANCE = 'track_performance';
+  private readonly PERFORMANCE_PREFIX = 'perf:';
+  private readonly APDEX_SUBFIX = ':apdex';
```

Comment:
I find it complicated when the separator is part of the value here because if you want to add or change the telemetry name it's easy to put the `:` in the wrong place.

My suggestion is to remove the `:` and instead of concatenating the string use join, eg:

```
[this.PERFORMANCE_PREFIX, 'app', this.APDEX_SUBFIX, this.APDEX_AGGREGATE].join(':');
```